velox/functions/prestosql/Reduce.cpp

```diff
@@ -67,39 +67,47 @@ class ReduceFunction : public exec::VectorFunction {
   void apply(
       const SelectivityVector& rows,
       std::vector<VectorPtr>& args,
-      const TypePtr& /* outputType */,
+      const TypePtr& outputType,
       exec::EvalCtx& context,
       VectorPtr& result) const override {
     VELOX_CHECK_EQ(args.size(), 4);
-
+    // Identify the rows need to be computed.
+    exec::LocalSelectivityVector nonNullRowsHolder(*context.execCtx());
+    const SelectivityVector* nonNullRows = &rows;
+    DecodedVector decodedArray;
+    decodedArray.decode(*args[0], rows);
+    if (decodedArray.mayHaveNulls()) {
+      nonNullRowsHolder.get(rows);
+      nonNullRowsHolder->deselectNulls(
+          decodedArray.nulls(), rows.begin(), rows.end());
+      nonNullRows = nonNullRowsHolder.get();
+      auto selected = nonNullRows->countSelected();
```

Comment:
can you please remove these unused variables?

Reply:
Sorry seems I pushed a dirty commit by mistake.